repository/repository.go

```diff
@@ -1,24 +1,77 @@
 package repository
 
-import "api/models"
+import (
+	"api/models"
+	"errors"
+	"fmt"
+	"strconv"
+)
+
+var books []models.Book
 
 type BookRepository interface {
 	GetAllBooks() ([]models.Book, error)
+	AddBook(book *models.Book) error
+	GetBookById(id string) (*models.Book, error)
+	UpdateBook(id string, updatedBook *models.Book) error
+	DeleteBook(id string) error
+}
+
+type inMemoryBookRepository struct {
+}
+
+func NewInMemoryBookRepository() *inMemoryBookRepository {
+	// Initialize books if needed
+	books = []models.Book{
+		{ID: "1", Title: "In Search of Lost Time", Author: "Marcel Proust", Quantity: 2},
+		{ID: "2", Title: "The Great Gatsby", Author: "F. Scott Fitzgerald", Quantity: 5},
+		{ID: "3", Title: "War and Peace", Author: "Leo Tolstoy", Quantity: 6},
+	}
+	return &inMemoryBookRepository{}
+}
```

Comment:
what if `books` pull it out from this function?

Reply:
ok i'll initialize the book slice outside the function

Reply:
Do you have a reason for this change?

Reply:
the reason is because it is better to make the books to be a package-level variable so other function can use it as well and to increase overall code readability

Reply:
is there anything else to improve kang @iqbal167 ?